Explorer/Assets/DCL/Backpack/BackpackEquipStatusController.cs

```diff
@@ -126,5 +108,35 @@ private void UpdateAvatarInWorld(Profile profile)
             else
                 world.Add(playerEntity.Value, profile);
         }
+
+        private void OnWearableEquipped(IWearable wearable)
+        {
+            equippedWearables.Equip(wearable);
+
+            async UniTaskVoid DisengageUnEquipIncompatibleWearablesAsync(IWearable bodyShape, CancellationToken ct)
+            {
+                // We need to wait to un-equip the rest of the wearables, otherwise the avatar and the UI is not updated correctly
+                await UniTask.NextFrame(ct);
```

Comment:
What's the flow here? It's quite dangerous just to skip a frame (looks like magic) without understanding the flow. Maybe it can be streamlined?

Reply:
Fixed here: https://github.com/decentraland/unity-explorer/pull/644/commits/02b836b75a4b462e28d1a9083e5c476cb98750ce